File: app/src/main/java/com/example/amit/uniconnexample/Fragment/Msgfrag.java
package com.example.amit.uniconnexample.Fragment;

import android.content.Context;
import android.content.Intent;
import android.net.ConnectivityManager;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.support.v4.widget.SwipeRefreshLayout;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;
import android.widget.Toast;

import com.example.amit.uniconnexample.Activity.Chatstart;
import com.example.amit.uniconnexample.Message_model;
import com.example.amit.uniconnexample.Activity.NewTabActivity;
import com.example.amit.uniconnexample.R;
import com.example.amit.uniconnexample.utils.Utils;
import com.firebase.ui.database.FirebaseRecyclerAdapter;
import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.database.DatabaseReference;
import com.google.firebase.database.FirebaseDatabase;

import java.util.ArrayList;

/**
 * Created by amit on 18/2/17.
 */

public class Msgfrag extends Fragment {
    RecyclerView rview;
    private DatabaseReference newsnd;
    SwipeRefreshLayout refresh;
    private FirebaseAuth auth;
    public Msgfrag() {
    }

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View view=inflater.inflate(R.layout.activity_msg_frag,container,false);
        rview = (RecyclerView)view. findViewById(R.id.mchat_list);
        refresh=(SwipeRefreshLayout)view.findViewById(R.id.refresh);
        //  FirebaseDatabase.getInstance().setPersistenceEnabled(true);
        auth=FirebaseAuth.getInstance();
        rview.setLayoutManager(new LinearLayoutManager(getActivity()));
        newsnd=FirebaseDatabase.getInstance().getReference().child("Smessage").child(auth.getCurrentUser().getUid());
        newsnd.keepSynced(true);
        refresh.setOnRefreshListener(new SwipeRefreshLayout.OnRefreshListener() {
            @Override
            public void onRefresh() {
                refresh();
            }
        });
        // if(isNetworkConnected()) {
        FirebaseRecyclerAdapter<Message_model, MessageViewHolder> firebaseRecyclerAdapter = new FirebaseRecyclerAdapter<Message_model, MessageViewHolder>(
                Message_model.class,
                R.layout.activity_messageitem,
                MessageViewHolder.class,
                newsnd
        ) {
            @Override
            protected void populateViewHolder(MessageViewHolder viewHolder, Message_model model, int position) {
                final String msg_key = getRef(position).getKey();
                //    Toast.makeText(Message.this,msg_key,Toast.LENGTH_LONG).show();
                // mDatabase.add
                viewHolder.bindData(model);
                viewHolder.view.setOnClickListener(new View.OnClickListener() {
                    @Override
                    public void onClick(View v) {
                        if(isNetworkConnected()) {
                            Intent i = new Intent(getActivity(), Chatstart.class);
                            i.putExtra("chat", msg_key);
                            startActivity(i);
                          // getActivity(). finish();
                        }else{
                            Toast.makeText(getActivity(),"No internet connection",Toast.LENGTH_LONG).show();
                        }
                    }
                });
            }
        };
        rview.setAdapter(firebaseRecyclerAdapter);
        return view;
    }

    public void refresh(){
        FirebaseRecyclerAdapter<Message_model, MessageViewHolder> firebaseRecyclerAdapter = new FirebaseRecyclerAdapter<Message_model, MessageViewHolder>(
                Message_model.class,
                R.layout.activity_messageitem,
                MessageViewHolder.class,
                newsnd
        ) {
            @Override
            protected void populateViewHolder(MessageViewHolder viewHolder, Message_model model, int position) {
                final String msg_key = getRef(position).getKey();
                //    Toast.makeText(Message.this,msg_key,Toast.LENGTH_LONG).show();
                // mDatabase.add
                viewHolder.bindData(model);
                viewHolder.view.setOnClickListener(new View.OnClickListener() {
                    @Override
                    public void onClick(View v) {
                        if(isNetworkConnected()) {
                            Intent i = new Intent(getActivity(), Chatstart.class);
                            i.putExtra("chat", msg_key);
                            startActivity(i);
                          //  getActivity().finish();
                        }else{
                            Toast.makeText(getActivity(),"No internet connection",Toast.LENGTH_LONG).show();
                        }
                    }
                });
            }
        };
        rview.setAdapter(firebaseRecyclerAdapter);
        refreshcomplete();
    }
    public void refreshcomplete(){
        refresh.setRefreshing(false);
    }

    @Override
    public void onStart() {
        super.onStart();
    }

    @Override
    public void onResume() {
        super.onResume();
        ((NewTabActivity)getActivity()).setTitle("    Message     ");
    }

    public static class MessageViewHolder extends RecyclerView.ViewHolder{
        View view;
        public MessageViewHolder(View itemView) {
            super(itemView);
            view=itemView;
        }
        public void bindData(Message_model model){
            TextView tmsg=(TextView)view.findViewById(R.id.txt);
            TextView tname=(TextView)view.findViewById(R.id.txtname);
            ImageView iview=(ImageView)view.findViewById(R.id.photo);
            tmsg.setText(model.getMsg());
            tname.setText(model.getName());
            if(model.getImage()!=null)
                iview.setImageBitmap(Utils.decodeBase64(model.getImage()));
            else{
            }
        }
    }
    private boolean isNetworkConnected() {
        ConnectivityManager cm = (ConnectivityManager) getActivity().getSystemService(
                Context.CONNECTIVITY_SERVICE);
        return cm.getActiveNetworkInfo() != null;
    }
}

File: app/src/main/java/com/example/amit/uniconnexample/Fragment/Msgfrag.kt
package com.example.amit.uniconnexample.Fragment

import android.content.Context
import android.content.Intent
import android.net.ConnectivityManager
import android.os.Bundle
import androidx.fragment.app.Fragment
import androidx.swiperefreshlayout.widget.SwipeRefreshLayout
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import android.widget.Toast

import com.example.amit.uniconnexample.Activity.Chatstart
import com.example.amit.uniconnexample.Message_model
import com.example.amit.uniconnexample.Activity.NewTabActivity
import com.example.amit.uniconnexample.R
import com.example.amit.uniconnexample.utils.Utils
import com.firebase.ui.database.FirebaseRecyclerAdapter
import com.google.firebase.auth.FirebaseAuth
import com.google.firebase.database.DatabaseReference
import com.google.firebase.database.FirebaseDatabase
import kotlinx.android.synthetic.main.activity_msg_frag.view.*

/**
 * Created by amit on 18/2/17.
 */

class Msgfrag : Fragment() {
    private var newsnd: DatabaseReference? = null
    private var auth: FirebaseAuth? = null
    lateinit var msgView: View
    private val isNetworkConnected: Boolean
        get() {
            val cm = activity!!.getSystemService(
                    Context.CONNECTIVITY_SERVICE) as ConnectivityManager
            return cm.activeNetworkInfo != null
        }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        msgView = inflater.inflate(R.layout.activity_msg_frag, container, false)
        auth = FirebaseAuth.getInstance()
        msgView.mchat_list.layoutManager = LinearLayoutManager(activity)
        newsnd = FirebaseDatabase.getInstance().reference.child("Smessage").child(auth!!.currentUser!!.uid)
        newsnd!!.keepSynced(true)
        msgView.refresh.setOnRefreshListener { refresh() }
        // if(isNetworkConnected()) {
        val firebaseRecyclerAdapter = object : FirebaseRecyclerAdapter<Message_model, MessageViewHolder>(
                Message_model::class.java,
                R.layout.activity_messageitem,
                MessageViewHolder::class.java,
                newsnd
        ) {
            override fun populateViewHolder(viewHolder: MessageViewHolder, model: Message_model, position: Int) {
                val msg_key = getRef(position).key
                //    Toast.makeText(Message.this,msg_key,Toast.LENGTH_LONG).show();
                // mDatabase.add
                viewHolder.bindData(model)
                viewHolder.view.setOnClickListener {
                    if (isNetworkConnected) {
                        val i = Intent(activity, Chatstart::class.java)
                        i.putExtra("chat", msg_key)
                        startActivity(i)
                        // getActivity(). finish();
                    } else {
                        Toast.makeText(activity, "No internet connection", Toast.LENGTH_LONG).show()
                    }
                }
            }
        }
        msgView.mchat_list.adapter = firebaseRecyclerAdapter
        return view
    }

    fun refresh() {
        val firebaseRecyclerAdapter = object : FirebaseRecyclerAdapter<Message_model, MessageViewHolder>(
                Message_model::class.java,
                R.layout.activity_messageitem,
                MessageViewHolder::class.java,
                newsnd
        ) {
            override fun populateViewHolder(viewHolder: MessageViewHolder, model: Message_model, position: Int) {
                val msg_key = getRef(position).key
                //    Toast.makeText(Message.this,msg_key,Toast.LENGTH_LONG).show();
                // mDatabase.add
                viewHolder.bindData(model)
                viewHolder.view.setOnClickListener {
                    if (isNetworkConnected) {
                        val i = Intent(activity, Chatstart::class.java)
                        i.putExtra("chat", msg_key)
                        startActivity(i)
                        //  getActivity().finish();
                    } else {
                        Toast.makeText(activity, "No internet connection", Toast.LENGTH_LONG).show()
                    }
                }
            }
        }
        msgView.mchat_list.adapter = firebaseRecyclerAdapter
        refreshcomplete()
    }

    fun refreshcomplete() {
        msgView.refresh.isRefreshing = false
    }

    override fun onStart() {
        super.onStart()
    }

    override fun onResume() {
        super.onResume()
        (activity as NewTabActivity).setTitle("    Message     ")
    }

    class MessageViewHolder(internal var view: View) : RecyclerView.ViewHolder(view) {
        fun bindData(model: Message_model) {
            val tmsg = view.findViewById<View>(R.id.txt) as TextView
            val tname = view.findViewById<View>(R.id.txtname) as TextView
            val iview = view.findViewById<View>(R.id.photo) as ImageView
            tmsg.setText(model.msg)
            tname.setText(model.name)
            if (model.image != null)
                iview.setImageBitmap(Utils.decodeBase64(model.image?:""))
            else {
            }
        }
    }
}
